test/jdk/java/lang/StableValue/StableValueTest.java

```diff
@@ -0,0 +1,389 @@
+                if (CACHED_KEYS.contains(a)) {
+                    return LOG2_CACHED.apply(a);
+                } else {
+                    return LOG2_ORIGINAL.apply(a);
+                }
+            }
+
+        }
+
+        double log16 = Log2Util.log2(16); // May eventually constant fold to 4.0 at runtime
+        double log256 = Log2Util.log2(256); // Will not constant fold
+
+        assertEquals(4, log16);
+        assertEquals(8, log256);
+    }
+
+    @Test
+    void functionExample2() {
+
+        class Log2Util {
+
+            private Log2Util() {}
+
+            private static final Set<Integer> KEYS =
+                    Set.of(1, 2, 4, 8);
+            private static final UnaryOperator<Integer> LOG2_ORIGINAL =
+                    i -> 31 - Integer.numberOfLeadingZeros(i);
+
+            private static final Function<Integer, Integer> LOG2 =
+                    // @link substring="function" target="#function(Set,Function)" :
+                    StableValue.function(KEYS, LOG2_ORIGINAL);
+
+            public static double log2(int a) {
+                return LOG2.apply(a);
+            }
+
+        }
+
+        double log16 = Log2Util.log2(8); // May eventually constant fold to 3.0 at runtime
+
+        assertEquals(3, log16);
+        assertThrows(IllegalArgumentException.class, () -> Log2Util.log2(3));
+    }
+
+    private static final BiPredicate<StableValue<Integer>, Integer> TRY_SET = StableValue::trySet;
+    private static final BiPredicate<StableValue<Integer>, Integer> SET_OR_THROW = (s, i) -> {
+        try {
+            s.setOrThrow(i);
+            return true;
+        } catch (IllegalStateException e) {
+            return false;
+        }
+    };
+
+    @Test
+    void raceTrySet() {
+        race(TRY_SET);
+    }
+
+    @Test
+    void raceSetOrThrow() {
+        race(SET_OR_THROW);
+    }
+
+    @Test
+    void raceMixed() {
+        race((s, i) -> switch (i % 2) {
+            case 0 -> TRY_SET.test(s, i);
+            case 1 -> SET_OR_THROW.test(s, i);
+            default -> fail("should not reach here");
+        });
+    }
+
+    void race(BiPredicate<StableValue<Integer>, Integer> winnerPredicate) {
+        int noThreads = 10;
+        CountDownLatch starter = new CountDownLatch(1);
+        StableValue<Integer> stable = StableValue.of();
+        Map<Integer, Boolean> winners = new ConcurrentHashMap<>();
+        List<Thread> threads = IntStream.range(0, noThreads).mapToObj(i -> new Thread(() -> {
+                    try {
+                        // Ready, set ...
+                        starter.await();
+                        // Here we go!
+                        winners.put(i, winnerPredicate.test(stable, i));
+                    } catch (Throwable t) {
+                        fail(t);
+                    }
+                }))
+                .toList();
+        threads.forEach(Thread::start);
+        LockSupport.parkNanos(TimeUnit.MILLISECONDS.toNanos(1));
```

Comment:
Instead of parking and calling `countDown` here, which might be unreliable if a particular thread doesn't get scheduled, you could set the latch value to `noThreads`, and then do a `starter.countDown()` before calling `await`. That should start all the threads running at the same time once all of them are ready.